### PR TITLE
Add dogfood cloud

### PR DESCRIPTION
**Pull request by danieljurek**

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]

File: eng/common/TestResources/New-TestResources.ps1 [hunk 1]
```diff
@@ -55,7 +55,6 @@ param (
     [string] $Location = '',
 
     [Parameter()]
-    [ValidateSet('AzureCloud', 'AzureUSGovernment', 'AzureChinaCloud')]
```
Comment (heaths) [inline eng/common/TestResources/New-TestResources.ps1:58]: I don't think you should get rid of these. You can add to the set, but they also help callers know which values are valid. You can type `-env` in the environment, hit tab a couple of times, and it shows the completion set. This helps usability.

File: eng/common/TestResources/deploy-test-resources.yml [hunk 2]
```diff
@@ -12,15 +12,13 @@ parameters:
 #   "TenantId": "<tenant id>",
 #   "TestApplicationId": "<test app id>",
 #   "TestApplicationSecret": "<test app secret>",
-#   "ProvisionerApplicationId": "<provisioner app id>",
-#   "ProvisionerApplicationSecret": "<provisioner app secret>",
+#   "ProvisionerApplicationId": "<provisoner app id>",
```
Comment (heaths) [inline eng/common/TestResources/deploy-test-resources.yml:15]: Everything was spelled right before. And you're not consistent here or below. "secert" is also spelled wrong.

File: eng/common/TestResources/setup-az-modules.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+steps:
+  - bash: sudo chown -R runner ~/.Azure
+    displayName: (MacOS) Grant access to ~/.Azure
+    condition: contains(variables['OSVmImage'], 'mac')
+
+  # New-TestResources command requires Az module
+  - pwsh: Install-Module -Name Az -Scope CurrentUser -AllowClobber -Force -Verbose
+    displayName: Install Azure PowerShell module
+
+  - task: Powershell@2
```
Comment (heaths) [inline eng/common/TestResources/setup-az-modules.yml:27]: Why not use the `pwsh` task like you did above? More concise and consistent.

**Review by heaths — CHANGES_REQUESTED** [review]

File: eng/common/TestResources/setup-az-modules.yml (hunk 3 above)
Comment (weshaggard) [reply eng/common/TestResources/setup-az-modules.yml:27]: For inline scripts like this yes I agree using pwsh works well. For one-line script calls we generally try to use the Powershell task directly as it is easier to diagnose and get the exact command line for debugging and trying to run it locally. FYI I've written up our guidance around this at https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/2/Azure-DevOps-Pipeline-Guidance?anchor=powershell-tips

File: eng/common/TestResources/New-TestResources.ps1 (hunk 1 above)
Comment (weshaggard) [reply eng/common/TestResources/New-TestResources.ps1:58]: Is there a way to have a known set but allow other un-validated options as well?

Comment (heaths) [reply eng/common/TestResources/New-TestResources.ps1:58]: Not that I'm aware. If adding new environments regularly is enough of a concern, please disregard my suggestion here.

**Review by mitchdenny — APPROVED** [review]

File: eng/common/TestResources/New-TestResources.ps1 (hunk 1 above)
Comment (damirault) [reply eng/common/TestResources/New-TestResources.ps1:58]: Readded validation set, with Dogfood now.

**Review by heaths — APPROVED** [review]

**Review by weshaggard — APPROVED** [review]

**Review by mitchdenny — APPROVED** [review]

**Pull request merged**